src/index.js

```diff
@@ -156,161 +168,139 @@ app.get('/travis.org/*', (req, res, next) => {
+    const query = { style: req.query.style };
 
-  console.log(`Incoming request from referrer: ${req.get('Referrer')}`);
+    const build = await travis.getLatestBranchBuild(branch);
+    const filters = {
+      env: req.query.env
+    };
+    const jobs = travis.filterJobs(build.jobs, filters);
+    const status = travis.aggregateStatus(jobs);
+    const color =
+      {
+        passed: 'brightgreen',
+        failed: 'red'
+      }[status] || 'lightgrey';
 
-  const user = req.params.user;
-  const repo = req.params.repo;
-  const endpoint = res.locals.travisEndpoint || undefined;
-  const branch = req.query.branch || 'master';
-  const label = req.query.label || req.params.repo;
-  const travis = new TravisClient(user, repo, endpoint);
-  const query = { style: req.query.style };
-  // eslint-disable-next-line promise/catch-or-return
-  travis
-    .getLatestBranchBuild(branch)
-    .then(build => {
-      const filters = {
-        env: req.query.env
-      };
-      const jobs = travis.filterJobs(build.jobs, filters);
-      const status = travis.aggregateStatus(jobs);
-      const color =
-        {
-          passed: 'brightgreen',
-          failed: 'red'
-        }[status] || 'lightgrey';
-      return getShieldsBadge(label, status, color, query);
-    })
-    .catch(err => {
+    let body;
+    try {
+      body = getShieldsBadge(label, status, color, query);
+    } catch (err) {
       console.error(`Error: ${err}`);
-      return getShieldsBadge(label, 'error', 'lightgrey', query);
-    })
-    // eslint-disable-next-line promise/always-return
-    .then(body => {
-      res.write(body);
-      res.end();
-    });
-});
+      body = getShieldsBadge(label, 'error', 'lightgrey', query);
+    }
 
-app.get('/travis/:user/:repo/sauce/:sauceUser?', (req, res) => {
-  res.status(200);
-  res.set('Content-Type', 'image/svg+xml');
-  if (cachingEnabled) {
-    res.set('Cache-Control', cacheHeader);
-  }
-  res.flushHeaders();
+    sendResponse({ res, body });
+  })
+);
 
-  console.log(`Incoming request from referrer: ${req.get('Referrer')}`);
+app.get(
+  '/travis/:user/:repo/sauce/:sauceUser?',
+  asyncMiddleware(async (req, res) => {
+    const { user, repo } = req.params;
+    const endpoint = res.locals.travisEndpoint || undefined;
+    const sauceUser = req.params.sauceUser || user;
+    const branch = req.query.branch || 'master';
 
-  const user = req.params.user;
-  const repo = req.params.repo;
-  const endpoint = res.locals.travisEndpoint || undefined;
-  const sauceUser = req.params.sauceUser || user;
-  const branch = req.query.branch || 'master';
-  const travis = new TravisClient(user, repo, endpoint);
-  // eslint-disable-next-line promise/catch-or-return
-  getSecret('sauce-access-key').then(accessKey => {
+    const accessKey = await getSecret('sauce-access-key');
+    const travis = new TravisClient(user, repo, endpoint);
+    const build = await travis.getLatestBranchBuild(branch);
```

Comment:
Would this be sped up at all by awaiting in parallel with `getSecret` above? Looks like `accessKey` isn't used until after this.

Reply:
Yep, I believe so!